pkg/controller/plan/adapter/ovirt/builder.go

```diff
@@ -394,9 +394,27 @@ func (r *Builder) mapCPU(vm *model.Workload, object *cnv.VirtualMachineSpec) {
 	}
 	if vm.CustomCpuModel != "" {
 		r.setCpuFlags(vm.CustomCpuModel, object)
+	} else if r.Plan.Spec.PreserveClusterCPUModel {
+		r.setCpuFlags(r.getClusterCpu(vm), object)
 	}
 }
 
+func (r *Builder) getClusterCpu(vm *model.Workload) string {
+	clusterVersion := vm.Cluster.Version.Major + "." + vm.Cluster.Version.Minor
+	for _, serverCpuOption := range vm.ServerCpu.SystemOptionValue {
+		if serverCpuOption.Version != clusterVersion {
```

Comment:
I commented on this below as well - can't we do it differently so we'll get here only with the CPUs for the compatibility level of the cluster? it seems odd to query all the values and filtering them out here